src/core/cluster/quorum.rs

```diff
@@ -231,14 +231,38 @@ impl Cluster for Quorum {
 
     //todo check no data (no error)
     async fn get(&self, key: BobKey) -> GetResult {
-        let all_nodes = self.get_target_nodes(key);
-        trace!("target nodes: {:?}", all_nodes);
-        debug!("GET[{}]: Nodes for fan out: {:?}", key, &all_nodes);
-
-        // @TODO currently we send requests to all nodes, ignoring quorum value.
-        // In future we need to call local node in first place and then remaining nodes.
-        let results = Self::get_all(key, all_nodes, GetOptions::new_all()).await;
-        debug!("GET[{}] cluster ans: {:?}", key, results);
+        debug!("GET[{}] quorum cluster", key);
+        trace!("GET[{}] ~~~LOOKUP LOCAL NODE~~~", key);
+        let (vdisk_id, disk_path) = self.mapper.get_operation(key);
+        if let Some(path) = disk_path {
+            debug!("local node has vdisk replica, check local");
+            let op = Operation::new_local(vdisk_id, path);
+            match self.backend.get_local(key, op).await {
+                Ok(data) => {
+                    debug!("GET[{}] key found on local node", key);
+                    return Ok(data);
+                }
+                Err(e) => error!("local node backend returned error: {}", e),
+            }
+        }
+        trace!("GET[{}] ~~~LOOKUP LOCAL NODE ALIEN~~~", key);
+        debug!("GET[{}] check local alien dir", key);
+        let op = Operation::new_alien(vdisk_id);
+        match self.backend.get_local(key, op).await {
+            Ok(data) => {
+                debug!("GET[{}] key found in local node alien", key);
+                return Ok(data);
+            }
+            Err(e) => error!("local node backend returned error: {}", e),
+        };
+        trace!("GET[{}] ~~~LOOKUP REMOTE NODES~~~", key);
+        let local_node = self.mapper.local_node_name();
+        let target_nodes = self
+            .get_target_nodes(key)
+            .iter()
+            .filter(|node| node.name() != local_node);
+        // @TODO don't wait for all answers, return on first success
+        let results = Self::get_all(key, target_nodes, GetOptions::new_all()).await;
```

Comment:
Should it have force option as well as we set particular nodes list?

Reply:
[outdated] `force_node` flag is in `Operation` and is true in all backend operations after cluster